### ISSUE TITLE
Go routine leaks in tunnel healthcheck in api-server

### ISSUE THREAD

**Issue by fabioy** [kubernetes/kubernetes#24279]
If the API server can't establish the SSH tunnel to the node, it seems to leak the go routine establishing the connection. See many go routines leaked with this stack:

golang.org/x/crypto/ssh.DiscardRequests()

Seems related to https://github.com/golang/go/issues/15113

It seems related to periodic api-server crashes that we're observing.

**Comment by lavalamp** [comment]
This is after about 1 day of an apiserver running in this condition.

```
$ curl 127.0.0.1:8080/debug/pprof/goroutine?debug=2|grep goroutine -n1| grep -v goroutine |sed 's/^[0-9]\{1,7\}-//'|sort|grep -v "^$"|grep -v "^--$"|sed 's/0x[a-f0-9]\{1,16\}//g'|uniq -c 
     23 github.com/coreos/etcd/client.(*simpleHTTPClient).Do(, , , , , , , , , , ...)
      1 github.com/golang/glog.(*loggingT).flushDaemon()
     20 golang.org/x/crypto/ssh.(*Client).handleChannelOpens(, )
     20 golang.org/x/crypto/ssh.(*Client).handleGlobalRequests(, )
  16529 golang.org/x/crypto/ssh.DiscardRequests()
     20 golang.org/x/crypto/ssh.(*forwardList).handleChannels(, )
     20 golang.org/x/crypto/ssh.(*handshakeTransport).readPacket(, , , , , )
      1 k8s.io/kubernetes/pkg/apiserver.(*timeoutHandler).ServeHTTP(, , , )
    118 k8s.io/kubernetes/pkg/apiserver.(*WatchServer).ServeHTTP(, , , )
     29 k8s.io/kubernetes/pkg/client/cache.(*Reflector).watchHandler(, , , , , , , )
      1 k8s.io/kubernetes/pkg/ssh.func·004()
    118 k8s.io/kubernetes/pkg/storage.(*cacheWatcher).process(, , , )
     23 k8s.io/kubernetes/pkg/storage/etcd.(*etcdWatcher).translate()
     18 k8s.io/kubernetes/pkg/storage.func·013()
      7 k8s.io/kubernetes/pkg/util/wait.JitterUntil(, , , )
      1 net.func·019()
  16549 net/http.(*persistConn).readLoop()
  16561 net/http.(*persistConn).writeLoop()
    193 net.(*pollDesc).Wait(, , , )
      1 os/signal.loop()
      1 runtime/pprof.writeGoroutineStacks(, , , )
     23 sync.(*Cond).Wait()
```

**Comment by cjcullen** [comment]
Changed the title to reflect the source of the leak.

**Comment by cjcullen** [comment]
/close
This was fixed by #24487
